pallets/ds-maps/src/lib.rs

```diff
@@ -0,0 +1,183 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+use frame_support::{
+    codec::{Decode, Encode},
+    decl_error, decl_event, decl_module, decl_storage, dispatch, ensure,
+    weights::{Weight},
+    Parameter,
+};
+use frame_system::ensure_signed;
+use pallet_ds_accounts as accounts;
+use accounts::REGISTRAR_ROLE;
+
+mod default_weight;
+#[cfg(test)]
+mod mock;
+#[cfg(feature = "payment")]
+mod payment;
+#[cfg(test)]
+mod tests;
+
+//Not sure, are those derives required?
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum ZoneType {
+    /// Forbidden type zone
+    Red,
+    /// Available for safe flights
+    Green,
+    /// Owns zones
+    Parent,
+    // 
+}
+
+impl Default for ZoneType {
+    fn default() -> Self {
+        ZoneType::Green
+    }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, Default, Debug, PartialEq, Eq)]
+pub struct BoxCoordinates<CoordinateSize> {
+    coordinates: [CoordinateSize; 6],
+}
+impl<CoordinateSize> BoxCoordinates<CoordinateSize>{
+    pub fn new(coordinates: [CoordinateSize; 6]) -> Self {
+        BoxCoordinates{coordinates}
+    }
+}
```

Comment:
I'd say it needs better design.

Maybe, it worth declaring `Point3D` type first, then using it declare `Box3D` which includes 2 points.

`CoordinateSize` is an inappropriate name for this parameter. Size is supposed to be a number. This parameter is not a number, it is a type. So it should be smth like `CoordinateType` or better just short `Coord`.

Array of 6 numbers is not the best way to express a box in 3d space.